### PR TITLE
Refactor app version to the main title

### PR DESCRIPTION
**Pull request by RussKie**
* Extract the application title generation from FormBrowse into `AppTitleGenerator`.
* Move functionality dealing with repository description from FormBrowse into `RepositoryDescriptionProvider`

### CODE REVIEW

File: UnitTests/GitCommandsTests/AppTitleGeneratorTests.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+        [Test]
+        public void Generate_should_include_suplied_branch_without_braces()
+        {
+            string branchName = "feature/my_(test)_branch";
+            var title = _appTitleGenerator.Generate("a", true, "(" + branchName + ")");
+            title.Should().StartWith($"{ShortName} ({branchName}) - Git Extensions");
+        }
+
+        [Conditional("DEBUG")]
```
Comment (gerhardol) [inline UnitTests/GitCommandsTests/AppTitleGeneratorTests.cs]: Not working, need to use #if !DEBUG
Similar in RevisionDiffProviderTest

File: GitCommands/Repository/RepositoryShortNameProvider.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+
+        /// <summary>
+        /// Returns a short name for repository.
+        /// If the repository contains a description it is returned,
+        /// otherwise the last part of path is returned.
+        /// </summary>
+        /// <param name="repositoryDir">Path to repository.</param>
+        /// <returns>Short name for repository</returns>
+        public string Get(string repositoryDir)
```
Comment (gerhardol) [inline GitCommands/Repository/RepositoryShortNameProvider.cs]: This should be the repo description, not repo shortname.
.git/description does not seem to be cloned.

I consider the repo shortname as the name part of the repo. Used like that in #4208

File: GitCommands/Repository/RepositoryShortNameProvider.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        {
+            var dirInfo = new DirectoryInfo(repositoryDir);
+            if (!dirInfo.Exists)
+            {
+                return dirInfo.Name;
```
Comment (gerhardol) [inline GitCommands/Repository/RepositoryShortNameProvider.cs]: ?

**Comment by gerhardol** [comment]
code review, a few comments only

File: GitCommands/Repository/RepositoryShortNameProvider.cs (hunk 2 above)
Comment (RussKie) [reply GitCommands/Repository/RepositoryShortNameProvider.cs]: Good point. I took the original method's name without really looking at the underlying functionality.

**Comment by RussKie** [comment]
> code review

I like that :)

File: GitCommands/Repository/RepositoryShortNameProvider.cs (hunk 3 above)
Comment (RussKie) [reply GitCommands/Repository/RepositoryShortNameProvider.cs]: [As per MSDN](https://msdn.microsoft.com/en-us/library/system.io.directoryinfo.directoryinfo(v=vs.110).aspx#Anchor_2) _This constructor does not check if a directory exists. _

So if the supplied path is invalid or not available (repo on a removable or network drive) you still want to see the name for it.

File: UnitTests/GitCommandsTests/AppTitleGeneratorTests.cs (hunk 1 above)
Comment (RussKie) [reply UnitTests/GitCommandsTests/AppTitleGeneratorTests.cs]: hmm, I kind of expected that test to run in debug only mode... clearly I was wrong

Comment (RussKie) [reply UnitTests/GitCommandsTests/AppTitleGeneratorTests.cs]: Thinking about it now, I think it works as expected... unfortunately for me

**Pull request merged**